jadx-gui/src/main/java/jadx/gui/utils/search/SearchImpl.java

```diff
@@ -0,0 +1,54 @@
+package jadx.gui.utils.search;
+
+import java.util.regex.Matcher;
+
+import org.apache.commons.lang3.StringUtils;
+
+public class SearchImpl {
+
+	/*
+	 * Checks if searchArea matches the searched string found in searchSettings
+	 */
+	public static boolean isMatch(StringRef searchArea, final SearchSettings searchSettings) {
+		return isMatch(searchArea.toString(), searchSettings);
```

Comment:
All your  `isMatch` and `find` implementations are static and have an `SearchSettings searchSettings` argument. 
So effectively they are handled like a virtual method of `SearchSettings`. I assume you wanted make a split between the data model and the implementation. Was this the reason or are there other reasons for doing so? 

Because moving the code into `SearchSettings` would in the end allow to use `SearchSettings` as interface and provide for each major search type an own optimized implementation.

Reply:
Your correct in that my goal was to separate the data model from the implementation with the idea to build a settings class to store all options which can be expanded in the future if needed.  Originally, I was thinking about just passing the settings class into the existing isMatched functions and doing the comparison there as was originally the case.  Although, it appeared that codeIndex, simpleIndex, and the searchNext function all implement the same algorithm with minor changes so I just moved that into the searchImpl class rather than updating it in all 3 places.

I just submitted another commit combining SearchImpl and SearchSettings as you suggested and agree that it is cleaner than my original implementation.  I still need to implement a search interface as recommended so will be making another commit.

What did you have in mind with doing an optimized implementation for each search type? Are you recommending to create a search interface then implement it in the index classes rather than using a generic find function implemented in the searchSettings class which is what I currently have.